### PR TITLE
Exclude predocket appeals from update representative job

### PR DESCRIPTION
**Pull request by leikkisa**
Resolves [Slack thread](https://dsva.slack.com/archives/C01DFC41BPV/p1637770547081500)

### Description
The update appeal representative job was looking at all active appeals, but we do not want to start doing this until an appeal is docketed. So this excludes pre-docketed appeals from the job.

This will be followed up by deleting the erroneously created tasks. Will post a script below.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Go to ...

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

 BEFORE|AFTER
 ---|---

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

File: app/jobs/update_appellant_representation_job.rb [hunk 1]
```diff
@@ -76,6 +76,10 @@ def active_appeals
     Appeal.joins(:tasks).where("tasks.type = ? AND tasks.status NOT IN (?)", "RootTask", Task.closed_statuses)
   end
 
+  def active_docketed_appeals
+    active_appeals.where.not(id: Appeal.pre_docket.map(&:id))
```
Comment (leikkisa) [inline app/jobs/update_appellant_representation_job.rb:80]: Started off subtracting them, but we need it to return an activerecord relation, not an array.  Happy to hear if anyone has other ideas on subtracting scope.

File: app/jobs/update_appellant_representation_job.rb [hunk 2]
```diff
@@ -45,16 +45,16 @@ def appeals_to_update
...
     )
 
     [legacy_appeals, appeals].flatten
   end
 
   def retrieve_number_to_update
-    number_of_legacy_appeals = legacy_appeals_with_hearings.size
-    number_of_ama_appeals = active_appeals.size
+    number_of_legacy_appeals = legacy_appeals_with_hearings.length
+    number_of_ama_appeals = active_docketed_appeals.length
```
Comment (leikkisa) [inline app/jobs/update_appellant_representation_job.rb:57]: `.size` and `.count` return a hash when using scopes (bug?) but I looked it up online and `.length` works more reliably on the active record relation.

**Review by ajspotts — APPROVED** [review]
lgtm!

**Comment by leikkisa** [comment]
Just pasting code ran to remove the extra tasks:

```ruby
pdts=PreDocketTask.open
as=pdts.map(&:appeal)
ts=as.map(&:tasks).flatten
tsd=ts.select{|t| ["TrackVeteranTask", "InformalHearingPresentationTask"].include?(t.type)}
tsd.each(&:destroy!)
```

File: app/jobs/update_appellant_representation_job.rb (hunk 2 above)
Comment (n1zyy) [reply app/jobs/update_appellant_representation_job.rb:57]: I ran into this the other day as well!

I think the answer is that the scopes are, somewhere along the line, using a `group_by`, which is why you get odd results. See: https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/docket.rb#L25-L31

I ended up calling `.count.size` but yours looks more reasonable!

**Pull request merged**